### DISCUSSION TITLE
Windows server 2016 as Worker

### DISCUSSION THREAD

**Discussion by skdubey** [concourse/concourse#6574] [General]
Hi, 

I am very new to Concourse world and trying to run a POC with Concourse web on Linux and a worker as Windows server 2016 to build a .Net (v 3.5) based appliction. 
Going through an official documentation, I do see the process for adding worker but all examples are based on Linux and not Windows. Even couldn't find any resource to start with. 
Can you please direct me to right path for this?

**Comment by taylorsilva** [comment]
Hello! A little history, Concourse started with a focus on Linux containers, so most of the team's expertise and product investment has fallen on that part of the codebase. Windows and Darwin have not changed much since early versions of Concourse and therefore have not kept up with the development of native windows containers.

I made a PR to the docs that expands on this cavet a bit: https://github.com/concourse/docs/pull/405/files
> Windows containers are currently not supported and Darwin does not
      have native containers. Steps will run inside a temporary directory on the
      Windows/Darwin worker. Any dependencies needed for your tasks (e.g. git,
      .NET, golang, ssh) should be pre-installed on the worker. Windows/Darwin
      workers do not come with any resource types.

Furthermore, all resource types are linux container images, thererfore they can't run on windows. A Concourse cluster needs at least one linux worker to do anything with resources (unless all your pipelines contain zero resources and only tasks, but that would suck 😞 ). That's also why the [resource](https://concourse-ci.org/resources.html) and [resource-types](https://concourse-ci.org/resource-types.html) docs don't have a `platform` field like the [`task`](https://concourse-ci.org/tasks.html) step does.
